.github/workflows/examples.yml

```diff
@@ -0,0 +1,28 @@
+name: examples
+
+on:
+  push:
+    branches: [ main ]
+
+jobs:
+  examples:
+    name: Build and commit examples
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+
+    - name: Build examples and docs
+      run: |
+        make examples
+        make docs
+
+    - name: Commit changed files from build
+      uses: stefanzweifel/git-auto-commit-action@v4
```

Comment:
I had an initial reaction against auto-committing all changed files rather than specific paths or directories or glob patterns. I guess I feel like it increases the risk of undesired stuff ending up in version control? Not a position that I hold strongly and I think I can be convinced that this is a non-issue. 

Wondering if we should log changed files in the PR workflow for better visibility into what is happening in the PR workflow.

Reply:
I added a glob pattern for the max set of files we'd expect to change

Reply:
Given that this will be in a single commit, it seems easy to click on that and see the files changed. But open to ideas if logging is desired